lib/ohai/plugins/linux/lspci.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Author:: Joerg Herzinger <joerg.herzinger@oiml.at>
+# Author:: Phil Dibowitz <phil@ipom.com>
+# Copyright:: Copyright (c) 2011 GLOBAL 2000/Friends of the Earth Austria
+# Copyright:: Copyright (c) 2017 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Ohai.plugin(:Lspci) do
+  depends "platform"
+  provides "pci"
+
+  collect_data(:linux) do
+    $devices = Mash.new
+    lspci = shell_out("lspci -vnnmk")
+
+    $h = /[0-9a-fA-F]/ #any hex digit
```

Comment:
Again re: global variables. Can we use instance variables, or define methods that return these regex patterns?